src/zenml/stack_stores/sql_stack_store.py

```diff
@@ -273,27 +361,55 @@ def deregister_stack(self, name: str) -> None:
 
     # Private interface implementations:
 
-    def _create_stack(
-        self, name: str, stack_configuration: Dict[StackComponentType, str]
+    def _save_stack(
+        self,
+        name: str,
+        stack_configuration: Dict[StackComponentType, str],
+        is_update: bool = False,
     ) -> None:
-        """Add a stack to storage.
+        """Save a stack.
 
         Args:
             name: The name to save the stack as.
             stack_configuration: Dict[StackComponentType, str] to persist.
         """
         with Session(self.engine) as session:
-            stack = ZenStack(name=name, created_by=1)
-            session.add(stack)
-            for ctype, cname in stack_configuration.items():
-                if cname is not None:
-                    session.add(
-                        ZenStackDefinition(
-                            stack_name=name,
-                            component_type=ctype,
-                            component_name=cname,
+            if is_update:
```

Comment:
I think it would be easier if you always treat this as a "create-or-update" call: first remove all definitions where `stack_name==name` and then re-create the whole set again from the input list. This way, you can completely ignore the input `is_update` flag and the implementation will look a lot simpler.

Reply:
Fixed.